### ISSUE TITLE
cpu.js Speed & Brand detection doesn't work with AMD processors

### ISSUE THREAD

**Issue by ItsEcholot** [sebhildebrandt/systeminformation#64]
```
[path]\node_modules\systeminformation\lib\cpu.js:165
            result.speed = name.split('@')[1].trim();
                                             ^

TypeError: Cannot read property 'trim' of undefined
    at [path]\node_modules\systeminformation\lib\cpu.js:165:46
    at ChildProcess.exithandler (child_process.js:197:7)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:194:7)
    at maybeClose (internal/child_process.js:899:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
```

This happens because the name string describing AMD processors doesn't contain the necessary data.
Sample name: 'AMD FX(tm)-8320 Eight-Core Processor'

**Comment by sebhildebrandt** [comment]
Yes, this is indeed a bug ... will have a look on it and provide an update end of week.

**Comment by ItsEcholot** [comment]
If you need someone with an AMD machine available to test stuff for you, don't hesitate to contact me.

**Comment by sebhildebrandt** [comment]
Great thank you. I am already working on a solution, will publish it tomorrow or Thursday. If you are able to test it then would be fine.

**Comment by ItsEcholot** [comment]
Sure, no worries

**Comment by sebhildebrandt** [comment]
Should work now

**Comment by ItsEcholot** [comment]
Awesome! Will test it today in the evening. 
Looks like a good workaround only thing I'm slightly worried about is that the speed value could be false if the processor has been overclocked as it's being set equal to speedmax. Then the speed value would no longer represent the correct factory speed like it does on Intel cpus. I however also have no idea for a better fix so I guess we have to deal with it. Maybe add some information to the docs.

I will confirm my suspicion today and report back with some facts.

**Comment by sebhildebrandt** [comment]
Yes, you are right. I will make further investigations how to get the default clock speed … didn’t find it yesterday

**Comment by ItsEcholot** [comment]
I'm not sure if it's possible to extract the default clock speed without a lookup table that includes the most common AMD cpus and their clock speed or something along these lines... I know this is a bad solution and I'm not sure how many people actually need the default clock speed of the cpu anyway so maybe just a notice in the documentation would be enough.

**Comment by sebhildebrandt** [comment]
Maybe the windows registry could help:

can you post the output of the following command (from commandline)

`REG QUERY HKEY_LOCAL_MACHINE\HARDWARE\DESCRIPTION\System\CentralProcessor\0`

**Comment by ItsEcholot** [comment]
Currently don't have access to an AMD machine but will test this in the evening. 
Chances are that the values available there are the same that are available through the Windows Management Instrumentation Commandline but it's sure worth a try.

**Comment by ItsEcholot** [comment]
No luck the registry reports the following:
```
HKEY_LOCAL_MACHINE\HARDWARE\DESCRIPTION\System\CentralProcessor\0
    Component Information    REG_BINARY    00000000000000000000000000000000
    Identifier    REG_SZ    AMD64 Family 21 Model 2 Stepping 0
    Configuration Data    REG_FULL_RESOURCE_DESCRIPTOR    FFFFFFFFFFFFFFFF0000000000000000
    ProcessorNameString    REG_SZ    AMD FX(tm)-8320 Eight-Core Processor
    VendorIdentifier    REG_SZ    AuthenticAMD
    FeatureSet    REG_DWORD    0x2c3b3dfe
    ~MHz    REG_DWORD    0x113f
    Update Revision    REG_DWORD    0x6000822
    Update Status    REG_SZ    Newer Patch Not Available
    Previous Update Revision    REG_DWORD    0x6000822
    Platform Specific Field1    REG_DWORD    0x6000822
```

The ~MHz entry resolves to 4415MHz which is the overclocked value on this machine.

**Comment by sebhildebrandt** [comment]
... working on a AMD list ... maybe this solves this issue ;-)

**Comment by sebhildebrandt** [comment]
Just published a version (3.32.4.) that should give you the correct base speed for AMD processors (using a lookup table). Maybe you have time to test it.

**Comment by ItsEcholot** [comment]
@sebhildebrandt Awesome! Will look into it in the next few days.
